eclib/JModel.ec

```diff
@@ -964,6 +964,30 @@ abbrev [-printing] VAESENCLAST      = AESENCLAST.
 abbrev [-printing] VAESIMC          = AESIMC.
 abbrev [-printing] VAESKEYGENASSIST = AESKEYGENASSIST.
 
+(* ------------------------------------------------------------------- *)
+(* PCLMULQDQ instructions *)
+(*
+| PCLMULQDQ
+| VPCLMULQDQ  of wsize
+*)
+op clmulq (x y: W64.t): W128.t =
+ let x128 =  W128.of_int (to_uint x) in
+ foldr (fun i r => (if y.[i] then x128 `<<<` i else W128.zero) `^` r)
+       W128.zero
+       (iota_ 0 64).
+
+op PCLMULQDQ (v1 v2: W128.t) (k: int): W128.t =
+ let x1 = v1 \bits64 (k %% 2) in
+ let x2 = v2 \bits64 (k %% 16 %% 2) in
```

Comment:
This line is suspicious, since k %% 16 %% 2 = k %% 2
 I think k%%16 should be k%/ 16.
I that case (k%%2) will be bit 0 of k and k %/ 16 %2 bit 4 of k.
Also why k is an int I was expected a W8.t.
But this k should be an immediate it is maybe not important.

Reply:
In fact, I think the extraction will fail. Because 
In jasmin the last argument is a w8 and in EC the last argument is a W8.t.

Reply:
yes, sorry about that. The index computation was intended to be (_ %/ _ %% _) -- but the whole EC portion was a last minute addition, and definitely very unsatisfactory :-(. I'll properly revise and test it soon.